### PR TITLE
Add support for `confirmation_method` on PaymentIntent

### PR DESCRIPTION
**Pull request by remi-stripe**
r? @mickjermsurawong-stripe 
cc @stripe/api-libraries @garrett-stripe

### CODE REVIEW

File: paymentintent.go [hunk 1]
```diff
@@ -90,8 +93,9 @@ type PaymentIntentParams struct {
 	Params               `form:"*"`
 	Amount               *int64                           `form:"amount"`
 	ApplicationFeeAmount *int64                           `form:"application_fee_amount"`
-	Confirm              *bool                            `form:"confirm"`
 	CaptureMethod        *string                          `form:"capture_method"`
+	Confirm              *bool                            `form:"confirm"`
+	ConfirmationMethod   *string                          `form:"confirmation_method"`
```
Comment (mickjermsurawong-stripe) [inline paymentintent.go]: @remi, I think `confirmation_method` is on `PaymentIntentCreateParams` instead?
It's currently gated in `enable_payment_intent_manual_confirmation`

**Comment by mickjermsurawong-stripe** [comment]
ptal @remi-stripe

File: paymentintent.go (hunk 1 above)
Comment (remi-stripe) [reply paymentintent.go]: @mickjermsurawong-stripe That's where I put it though https://github.com/stripe/stripe-go/blob/5335e5f4161ab7422d39b7fe1f9983121584bbe8/paymentintent.go#L98

We don't have CreateParams and UpdateParams in go. Just Params. So it works, it just only works on Create like CaptureMethod

Comment (mickjermsurawong-stripe) [reply paymentintent.go]: ah ack! sorry I forgot the convention we have here.

**Comment by mickjermsurawong-stripe** [comment]
LGTM

**Pull request merged**